src/main/java/org/broadinstitute/consent/http/service/LibraryCardService.java

```diff
@@ -112,16 +114,20 @@ public void removeDaaFromLibraryCard(Integer libraryCardId, Integer daaId) {
     libraryCardDAO.deleteLibraryCardDaaRelation(libraryCardId, daaId);
   }
 
-  public List<LibraryCard> addDaaToUserLibraryCardByInstitution(User user, Integer institutionId, Integer daaId) {
-    List<LibraryCard> libraryCards = findLibraryCardsByUserId(user.getUserId());
-    List<LibraryCard> matchingLibraryCards = libraryCards.stream()
-        .filter(card -> Objects.equals(card.getInstitutionId(), institutionId))
-        .toList();
+  public List<LibraryCard> addDaaToUserLibraryCardByInstitution(User user, User signingOfficial, Integer daaId) {
+    if (signingOfficial.getInstitutionId() == null) {
+      throw new BadRequestException("This signing official does not have an institution.");
```

Comment:
Could we add a test for these lines?

Reply:
I believe there is one: https://github.com/DataBiosphere/consent/pull/2327/files#diff-4fea547504587c95ef46d7936f68c193192046e49d4349cb3834ff2c9c1cabe0R455

Reply:
If you set a breakpoint on this line and run the test in debug mode it will skip these lines since the institution is set. It throws later, so I suspect the test is not running as expected.

To see coverage, click the button next to tests and select "Run <test> with Coverage".